TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -314,12 +314,19 @@ public virtual DataType PrimitiveDataType
             get
             {
                 if (_primitiveDataType != null) return _primitiveDataType;
-                if (this.Picture != null) //Get DataType based on Picture clause
-                    _primitiveDataType = DataType.Create(this.Picture.Value);
-                else if (this.Usage.HasValue) //Get DataType based on Usage clause
-                    _primitiveDataType = DataType.Create(this.Usage.Value);
+
+                if (DataType == DataType.Date || DataType == DataType.Currency)
+                {
+                    // special case for built-ins Date and Currency, we accept
+                    _primitiveDataType = DataType.Alphanumeric;
```

Comment:
Is is possible to set PrimitiveDataType of types like Date and Currency in method `public static Nodes.TypeDefinition CreateBuiltIn(DataType type)` of class Datatype ?
This will avoid the special if in this method.